scikeras/wrappers.py

```diff
@@ -465,109 +464,89 @@ def _check_array_dtype(arr):
-        ----------
-        y : np.ndarray
-            1D or 2D numpy array.
-        reset : bool, optional
-            [description], by default True
-
-        Returns
-        -------
-        y : Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]]
-            Transformed target that will be passed directly to Keras.
-        meta : Dict[str, Any]
-            Meta parameters, as determined from input.
-        """
-        if not reset:
-            if not np.can_cast(y.dtype, self.y_dtype_):
+            if len(X_shape_) != len(self.X_shape_):
                 raise ValueError(
-                    f"Got `y` with dtype {y.dtype},"
-                    f" but this {self.__name__} expected {self.X_dtype_}"
-                    f" and casting from {y.dtype} to {self.X_dtype_} is not safe!"
+                    f"`X` has {len(X_shape_)} dimensions, but this {self.__name__}"
+                    f" is expecting {len(self.X_shape_)} dimensions in `X`."
                 )
-            if len(self.y_shape_) != len(y.shape):
+            # The following check is a backport from
+            # sklearn.base.BaseEstimator._check_n_features
+            # since this method is not available in sklearn <= 0.22.0
+            if n_features_in_ != self.n_features_in_:
                 raise ValueError(
-                    f"`y` has {len(y.shape)} dimensions, but this {self.__name__}"
-                    f" is expecting {len(self.y_shape_)} dimensions in `y`."
+                    "X has {} features, but {} is expecting {} features "
+                    "as input.".format(
+                        n_features_in_, self.__class__.__name__, self.n_features_in_
+                    )
                 )
-        meta = {
-            "y_dtype_": y.dtype,
-            "y_shape_": y.shape,
-        }
-        return y, meta
 
-    def postprocess_y(
-        self, y: Union[np.ndarray, List[np.ndarray]], reset: bool = True
-    ) -> np.ndarray:
-        """Handles manipulation of predicted `y` values.
+        if y is None:
+            return X
+        return X, y
 
-        By default, it joins lists of predictions for multi-output models
-        into a single numpy array.
-        Override this method to customize processing.
+    @property
+    def target_encoder(self) -> BaseKerasTransformer:
+        """Retrieve a transformer for targets / ``y``.
 
-        Arguments:
-            y : 2D numpy array or list of numpy arrays
-                (the latter is for multi-output models)
+        Metadata will be collected from `get_metadata` if
+        the transformer implements that method.
 
-        Returns:
-            y : 2D numpy array with singular dimensions stripped
-                or 1D numpy array
+        Returns
+        -------
+        BaseKerasTransformer
+            Transformer implementing the BaseKerasTransformer
+            interface.
         """
-        return np.squeeze(np.column_stack(y))
+        raise NotImplementedError
```

Comment:
I think by default, BaseWrapper shouldn't do any data processing instead of raising a ValueError.

Reply:
That's what I originally had, I'll revert, not a huge deal to me.

Reply:
Done